Chewbawka2017/src/org/team1540/chewbawka/GearSlider.java

```diff
@@ -1,7 +1,98 @@
 package org.team1540.chewbawka;
 
-public class GearSlider {
-	public static void setup() {
+import ccre.channel.BooleanCell;
+import ccre.channel.BooleanInput;
+import ccre.channel.EventCell;
+import ccre.channel.EventOutput;
+import ccre.channel.FloatCell;
+import ccre.channel.FloatIO;
+import ccre.channel.FloatInput;
+import ccre.channel.FloatOutput;
+import ccre.cluck.Cluck;
+import ccre.ctrl.ExtendedMotor.OutputControlMode;
+import ccre.ctrl.ExtendedMotorFailureException;
+import ccre.ctrl.StateMachine;
+import ccre.drivers.ctre.talon.TalonExtendedMotor;
+import ccre.frc.FRC;
 
+public class GearSlider {
+	
+	public static final TalonExtendedMotor gearSliderMotor = FRC.talonCAN(17);
+	public static final FloatOutput servoLeft = FRC.servo(4, 0, 120);
+	public static final FloatOutput servoRight = FRC.servo(5, -120, 0);
+	// servo assumptions: positive = clockwise, 0 = down
+	
+	public static void setup() throws ExtendedMotorFailureException {
+		
+		BooleanCell depositingGear = new BooleanCell();
+		ControlBindings.gearServoButton.onPress(() -> depositingGear.invert());
+		depositingGear.onPress(() -> {servoLeft.set(0); servoRight.set(120);});
+		depositingGear.onRelease(() -> {servoLeft.set(120); servoLeft.set(0);});
+		
+		StateMachine gearSliderStates = new StateMachine(0,
+				"not calibrated",
+				"calibrating 0 end",
+				"calibrating distance",
+				"returning to center",
+				"calibrated");
+		
+		gearSliderMotor.modEncoder().configureEncoderCodesPerRev(125 * 15);
+		gearSliderMotor.modGeneralConfig().configureMaximumOutputVoltage(12f, -12f);
+		FloatCell calibratingSpeed = new FloatCell();
+		FloatOutput gearSliderSpeedControl = gearSliderMotor.asMode(OutputControlMode.SPEED_FIXED);
+		FloatOutput gearSliderPositionControl = gearSliderMotor.asMode(OutputControlMode.POSITION_FIXED);
+		FloatIO gearSliderPosition = gearSliderMotor.modEncoder().getEncoderPosition();
+		FloatInput gearSliderVelocity = gearSliderMotor.modEncoder().getEncoderVelocity();
+		
+		FloatCell calibrationSpeed = new FloatCell(10f);
+		FloatCell slow = new FloatCell(2f);
+		BooleanInput tooSlow = gearSliderVelocity.absolute().atMost(slow);
+		FloatCell slidingDistance = new FloatCell();
+		BooleanCell calibrated = new BooleanCell();
+		FloatCell positionError = new FloatCell(10f);
+		BooleanInput atMiddle = gearSliderPosition.inRange(slidingDistance.dividedBy(2f).minus(positionError), 
```

Comment:
It's somewhat confusing what's happening here (lines 53-82 basically, because of all the interactions between everything.) Some comments would be nice.